### PR TITLE
feat(java): bump version to beta.1

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1441](https://algolia.atlassian.net/browse/DI-1441)

Bump java version to `beta.0` !
And merge the changelog to this version

[DI-1441]: https://algolia.atlassian.net/browse/DI-1441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playground/java/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ repositories {
 
 dependencies {
     implementation 'io.github.cdimascio:dotenv-java:3.0.0'
-    implementation 'com.algolia:algoliasearch-client-java:0.0.1-SNAPSHOT'
+    implementation 'com.algolia:algoliasearch-client-java:4.0.0-beta.0'
```
Comment (aallam) [inline playground/java/build.gradle:12]: Is there a reason to start with beta `0` ? is this related to the release process ? if not, we can start directly with `1` !

**Review by aallam — COMMENTED** [review]
Nice! 🚀

File: playground/java/build.gradle (hunk 1 above)
Comment (Fluf22) [reply playground/java/build.gradle:12]: I think we always begin version number at 0, at Algolia 🤔

Comment (millotp) [reply playground/java/build.gradle:12]: I don't mind changing to 1, it was written 0 in the ticket ahah

Comment (shortcuts) [reply playground/java/build.gradle:12]: yep I think we always start new versions at 1

Comment (millotp) [reply playground/java/build.gradle]: changed to 1

**Comment by shortcuts** [comment]
can you update the PR title pls? so it's correct on the changelog @millotp
